Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -150,7 +163,8 @@ bool ScrollAnchoringController::didFindPriorityCandidate(Document& document)
 
         while (iterElement && iterElement.get() != elementForScrollableArea(m_owningScrollableArea)) {
             auto candidateResult = examineAnchorCandidate(*iterElement);
-            if (candidateResult == CandidateExaminationResult::Exclude || (iterElement == candidateElement && candidateResult == CandidateExaminationResult::Skip))
+            auto scrollAnchoringController = scrollAnchoringControllerForElement(*iterElement);
+            if (candidateResult == CandidateExaminationResult::Exclude || (iterElement == candidateElement && candidateResult == CandidateExaminationResult::Skip) || (scrollAnchoringController && scrollAnchoringController->anchorElement()))
```

Comment:
This logic is getting hard to follow; a lambda might be better. I'm not clear on why `scrollAnchoringController` having its own anchor means we return null here. In general, be careful about "reaching across" into another object; we're in `ScrollAnchoringController` code here, but it's doing something with a different `ScrollAnchoringController` and this gets confusing.

For example, how can I tell if `scrollAnchoringController` is for a descendant of `this`, or a sibling or ancestor? If there's code that needs to reason about multiple ScrollAnchoringControllers, consider putting it into a static function.

Reply:
Right this is getting a bit confusing. Separated this to a static function and added a few comments.